packages/@uppy/compressor/src/index.js

```diff
@@ -53,14 +53,20 @@ export default class Compressor extends BasePlugin {
           totalCompressedSize += compressedSavingsSize
           const { name, type, size } = compressedBlob
           const extension = name && getFileNameAndExtension(name).extension
+          // Name might have been changed in file.meta, so we update only the extension
+          const newMetaName = `${getFileNameAndExtension(file.meta.name).name}.${extension}`
+
           this.uppy.setFileState(file.id, {
             ...(name && { name }),
             ...(extension && { extension }),
             ...(type && { type }),
             ...(size && { size }),
```

Comment:
Wouldn't this be the same without doing a separate call?

```suggestion
            ...(size && { size }),
            meta: { ...file.meta, name: newMetaName }
```